lib/system/fs.go

```diff
@@ -48,3 +51,67 @@ func GetFilesystem(ctx context.Context, path string, runner utils.CommandRunner)
 
 	return "", trace.NotFound("no filesystem found for %v", path)
 }
+
+// RemoveFilesystem erases filesystem from the provided device/partition.
+func RemoveFilesystem(path string, log logrus.FieldLogger) error {
+	var out bytes.Buffer
+	// We don't need to fill the entire disk/partition with zeroes,
+	// clearing out the first 1MB should be enough to wipe out the
+	// filesystem header.
+	cmd := exec.Command("dd", "if=/dev/zero", fmt.Sprintf("of=%v", path), "bs=1M", "count=1")
+	if err := utils.ExecL(cmd, &out, log); err != nil {
+		return trace.Wrap(err, "failed to erase %v: %s", path, out.String())
+	}
+	return nil
+}
+
+// FormatDevice formats block device at the specified path to either xfs or,
+// if that fails, ext4, and returns the resulting filesystem.
```

Comment:
This function I also just moved as-is from elsewhere and reused. Do you suggest to just always format as `xfs`?

Reply:
IIRC, overlay(2) needs a specific backing filesystem - it cannot be arbitrary, so we might need to force-format if that's neither `xfs` nor `ext4`. Need to double check the filesystems usable as backing fs for overlay though.

Reply:
TBH it doesn't matter in this particular case b/c the device that we're dealing with during upgrade will be always unformatted anyway. But I added an option to force-reformat just in case.